test/models/shipit/webhooks/handlers/pull_request/closed_handler_test.rb

```diff
@@ -0,0 +1,192 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module Shipit
+  module Webhooks
+    module Handlers
+      module PullRequest
+        class ClosedHandlerTest < ActiveSupport::TestCase
+          test "validates payload" do
+            assert_raise(StandardError) { Shipit::Webhooks::Handlers::PullRequest::ClosedHandler.new(payload_parsed(:invalid_pull_request)) }
+          end
+
+          test "ignores irrelevant PR actions" do
+            assert_no_difference -> { Shipit::Stack.not_archived.count } do
+              Shipit::Webhooks::Handlers::PullRequest::ClosedHandler.new(payload_parsed(:pull_request_opened).merge(action: "assigned")).process
+            end
+          end
+
+          test "does not error for repos that are not tracked" do
+            Shipit::Webhooks::Handlers::PullRequest::ClosedHandler.new(payload_parsed(:pull_request_with_no_repo).merge(action: "closed")).process
+          end
+
+          test "archives stacks for repos that are tracked" do
+            create_stack
+            assert_difference -> { Shipit::Stack.not_archived.count }, -1 do
+              Shipit::Webhooks::Handlers::PullRequest::ClosedHandler.new(payload_parsed(:pull_request_closed)).process
+            end
+          end
+
+          test "ignored duplicate deliveries" do
+            Shipit::Webhooks::Handlers::PullRequest::ClosedHandler.new(payload_parsed(:pull_request_opened)).process
+            Shipit::Webhooks::Handlers::PullRequest::ClosedHandler.new(payload_parsed(:pull_request_closed)).process
```

Comment:
More of a nitpick, but I see many test using fully qualified class names when they could use much shorter ones.